images/manageiq-base/Dockerfile

```diff
@@ -56,115 +36,36 @@ LABEL name="manageiq-base" \
-      cyrus-sasl                  \
-      cyrus-sasl-plain            \
-      file                        \
-      freeipmi                    \
-      gcc-c++                     \
-      git                         \
-      glibc-langpack-en           \
-      hostname                    \
-      http-parser                 \
-      ipmitool                    \
-      iproute                     \
-      libcurl-devel               \
-      libpq                       \
-      libssh2-devel               \
-      libtool                     \
-      libxml2-devel               \
-      libxslt-devel               \
-      logrotate                   \
-      lvm2                        \
-      make                        \
-      net-snmp                    \
-      net-snmp-libs               \
-      net-snmp-utils              \
-      nmap-ncat                   \
-      nodejs                      \
-      openldap-clients            \
-      openscap-scanner            \
-      openssl-devel               \
-      OpenIPMI                    \
-      patch                       \
-      postfix                     \
-      postgresql-devel            \
-      procps-ng                   \
+      manageiq-pods               \
       python3-devel               \
-      qpid-proton-c-devel         \
-      redhat-rpm-config           \
-      ruby-devel                  \
-      socat                       \
-      sqlite-devel                \
-      v2v-conversion-host-ansible \
-      wget                        \
-      xz                          \
       &&                          \
     dnf clean all
 
-RUN if [ ${ARCH} != "ppc64le" ] ; then \
-    dnf -y --disableplugin=subscription-manager --setopt=tsflags=nodocs install wmi ; fi
-
 # Install python packages the same way the appliance does
 COPY --from=0 build/kickstarts/partials/post/python_modules.ks.erb /tmp/python_modules
 RUN bash /tmp/python_modules && rm -f /tmp/python_modules
 COPY --from=0 build/kickstarts/partials/post/ovirt_ansible.ks.erb /tmp/ovirt_ansible
 RUN bash /tmp/ovirt_ansible && rm -f /tmp/ovirt_ansible
 
-RUN mkdir -p ${APP_ROOT} ${APPLIANCE_ROOT} ${SUI_ROOT} && \
-    ln -vs ${APP_ROOT} /opt/manageiq/manageiq
-COPY --from=0 app ${APP_ROOT}
-COPY --from=0 appliance ${APPLIANCE_ROOT}
-COPY --from=0 sui ${SUI_ROOT}
-
-RUN ${APPLIANCE_ROOT}/setup && \
-    mkdir -p ${APP_ROOT}/log/apache && \
-    rm -f /etc/httpd/conf.d/* && \
-    cp ${APP_ROOT}/config/cable.yml.sample ${APP_ROOT}/config/cable.yml
-
-# build and install gems
-WORKDIR ${APP_ROOT}
-RUN source /etc/default/evm && \
-    export RAILS_USE_MEMORY_STORE="true" && \
-    gem install bundler -v ">=1.16.2" && \
-    if [ ${ARCH} = "ppc64le" ] ; then gem install sassc  -- --disable-march-tune-native && \
-        gem install unf_ext -v '0.0.7.2' -- --with-cxxflags="-fsigned-char"; fi && \
-    bundle install --with qpid_proton && \
-    rake tmp:clear evm:compile_sti_loader log:clear && \
```

Comment:
This line was actually creating the `/var/lib/data` directory at build time which is why we were not having this problem before.

Not sure if the right answer is to just dump a `rake environment` in here at build time or try to make the data directory logic conditional on if we're running in a container or not... Thoughts @Fryguy @jrafanie ?

Reply:
I guess it also depends on what we're using that directory for at runtime. If it's not in the source tree we might not have write permissions to it.

Reply:
What is this used for?  I was assuming the miq_tmp stuff was for the RHV fleecing code, but according to https://github.com/ManageIQ/manageiq-appliance-build/blob/master/kickstarts/partials/main/disk_layout.ks.erb#L7  it should be at /var/www/miq_tmp .  @simaishi @jrafanie ?

Even so, I don't think it actually matters where the data directory ultimately ends up, particularly in podified.  I'm not even sure the RHV fleecing code will work in podified, regardless.

Either way I feel like this directory should be precreated at build time with write permissions for the application.  So, I'm fine with a `rake environment`, or even a straight `RUN mkdir -p /var/lib/data` in the Dockerfile.

Reply:
@Fryguy I'm actually thinking it might make more sense to just add to [this conditional](https://github.com/ManageIQ/manageiq/blob/5996d45ed01c2d2f4cd662fd26ecc7f9e3218fc8/lib/vmdb_helper.rb#L26) so that it stays within the source tree. Then we don't have to fix the permissions.

Reply:
Neither of the locations are on a volume in pods, so it makes more sense for it to live in rails tmp directory.

Reply:
https://github.com/ManageIQ/manageiq/pull/20224 Fixes it